packages/types/src/utils/typeguards.ts

```diff
@@ -1,51 +1,67 @@
+
+export function isExecutionPayload<F extends ForkExecution>(
+  payload: ExecutionPayload<F> | BlindedExecutionPayload<F>
+): payload is ExecutionPayload<F> {
+  // we just check transactionsRoot for determining as it the base field
+  // that is present and differs from ExecutionPayload for all forks
+  return (payload as ExecutionPayload<F>).transactions !== undefined;
+}
 
-export function isBlindedExecution(payload: FullOrBlindedExecutionPayload): payload is ExecutionPayloadHeader {
-  // we just check transactionsRoot for determinging as it the base field
+export function isBlindedExecutionPayload<F extends ForkExecution>(
+  payload: ExecutionPayload<F> | BlindedExecutionPayload<F>
+): payload is BlindedExecutionPayload<F> {
+  // we just check transactionsRoot for determining as it the base field
   // that is present and differs from ExecutionPayload for all forks
-  return (payload as ExecutionPayloadHeader).transactionsRoot !== undefined;
+  return (payload as BlindedExecutionPayload<F>).transactionsRoot !== undefined;
 }
 
-export function isBlindedBeaconBlock(block: FullOrBlindedBeaconBlockOrContents): block is BlindedBeaconBlock {
-  const body = (block as FullOrBlindedBeaconBlock).body;
-  return body !== undefined && isBlindedBeaconBlockBody(body);
+export function isExecutionPayloadAndBlobsBundle<F extends ForkBlobs>(
+  data: ExecutionPayload<ForkExecution> | ExecutionPayloadAndBlobsBundle<F>
+): data is ExecutionPayloadAndBlobsBundle<F> {
+  return (data as ExecutionPayloadAndBlobsBundle<ForkBlobs>).blobsBundle !== undefined;
 }
 
-export function isBlindedBeaconBlockBody(body: FullOrBlindedBeaconBlockBody): body is BlindedBeaconBlockBody {
-  return (body as BlindedBeaconBlockBody).executionPayloadHeader !== undefined;
+export function isBlindedBeaconBlock<F extends ForkExecution>(
+  block: BeaconBlockOrContents | SignedBeaconBlockOrContents
+): block is BlindedBeaconBlock<F> {
+  return "body" in block && block.body !== undefined && block.body !== null && isBlindedBeaconBlockBody(block.body);
 }
 
-export function isBlindedSignedBeaconBlock(
-  signedBlock: FullOrBlindedSignedBeaconBlock
-): signedBlock is SignedBlindedBeaconBlock {
-  return (signedBlock as SignedBlindedBeaconBlock).message.body.executionPayloadHeader !== undefined;
+export function isBlindedSignedBeaconBlock<F extends ForkExecution>(
+  signedBlock: SignedBeaconBlock | SignedBeaconBlockOrContents
+): signedBlock is SignedBlindedBeaconBlock<F> {
+  return (signedBlock as SignedBlindedBeaconBlock<F>).message.body.executionPayloadHeader !== undefined;
 }
 
-export function isBlockContents(data: FullOrBlindedBeaconBlockOrContents): data is BlockContents {
-  return (data as BlockContents).kzgProofs !== undefined;
+export function isBlindedBeaconBlockBody<F extends ForkExecution>(
+  body: BeaconBlockBody | BlindedBeaconBlockBody
+): body is BlindedBeaconBlockBody<F> {
+  return "executionPayloadHeader" in body && body.executionPayloadHeader !== undefined;
```

Comment:
Because `as T` pattern code is and old one which I didn't touched. `in obj` is what I touched so improved it. For type guards when you have variant object best approach to is to use `in` to check for it's own property. 

We have to use `in` operator else the type will fail on next condition and.

Reply:
Updated the type guards.